### PR TITLE
Fix protobuf generation outside of GOPATH

### PR DESCRIPTION
**Pull request by azdagron**
Recent go modules work allows the SPIRE codebase to be checked out and built outside of a GOPATH. However, protobuf generation done by `build.sh` is dependent on the source living inside of a GOPATH. This PR fixes protobuf code generation by removing this dependency.

In summary:

- removed GOPATH relative import path from protoc invocation
- updated .proto files go_package declaration to be the full Go import path (necessary for proper code generation with go modules)
- fixed imports of local .proto's to be relative to the project root according to protobuf authoring guidelines
- added `paths=source_relative` option support to protoc-gen-spireplugin, along with other minor fixes
- updated grpc-gateway plugin to gain `paths=source_relative` support
- updated protoc invocations to pass `paths=source_relative` to plugins
- updated the GOPATH value used by build.sh to default to the same default as the `go` tool if no GOPATH is present (i.e. $HOME/go).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -21,7 +21,7 @@ require (
 	github.com/golang/protobuf v1.2.0
 	github.com/google/go-cmp v0.2.0 // indirect
 	github.com/gopherjs/gopherjs v0.0.0-20181103185306-d547d1d9531e // indirect
-	github.com/grpc-ecosystem/grpc-gateway v1.4.1
+	github.com/grpc-ecosystem/grpc-gateway v1.6.4
```
Comment (evan2645) [inline go.mod]: I don't think we use the grpc-gateway stuff anymore? Can we get rid of it?

File: tools/protoc-gen-spireplugin/main.go [hunk 2]
```diff
@@ -30,7 +32,11 @@ func main() {
 	req := new(plugin.CodeGeneratorRequest)
 	panice(proto.Unmarshal(reqData, req))
 
-	g := NewGenerator(req)
+	// parse parameters
+	options := parseOptions(req.GetParameter())
+	options = options
```
Comment (evan2645) [inline tools/protoc-gen-spireplugin/main.go]: Out of curiosity, what does this operation do?

Comment (azdagron) [reply tools/protoc-gen-spireplugin/main.go]: Oops. That's development cruft left to silence the "unused" compiler error (before i had plumbed through options). Will remove.

File: go.mod (hunk 1 above)
Comment (azdagron) [reply go.mod]: I'd be super happy to get rid of it :)

**Comment by azdagron** [comment]
From a consumer standpoint, the go_package change should be benign; it doesn't impact the generated protobuf code. It does give more information to plugins so they don't have to try and guess the Go import path based on the source path. As far as the .proto imports...., it may impact the code generation step in that an import path into github.com/spiffe/spire will be necessary. The [proto3 language guide](https://developers.google.com/protocol-buffers/docs/proto3#importing-definitions) recommends that all imports are done from the root of the project. The import "path" from the root of our project starts with "proto" which isn't very clear. I wonder if we shouldn't add "spire" and "spiffe" directories inside of "proto". Then import paths would be something like "spire/api/node.proto" or "spiffe/api/workload.proto" all available under a single protobuf import path.

**Comment by azdagron** [comment]
#709 took care of the immediate need of fixing protobuf generation/verification so that it works when SPIRE is checked out inside, and outside, of a GOPATH.

The other changes in here, while I believe are positive in terms of protobuf best practices, are breaking changes and unnecessary at this time.

**Pull request closed without merging**